Kernel/Firmware/ACPI/StaticParsing.cpp

```diff
@@ -31,16 +32,35 @@ Optional<PhysicalAddress> find_rsdp()
 
 static bool match_table_signature(PhysicalAddress table_header, StringView signature)
 {
-    // FIXME: There's no validation of ACPI tables here. Use the checksum to validate the tables.
     VERIFY(signature.length() == 4);
 
-    auto table = Memory::map_typed<Structures::RSDT>(table_header).release_value_but_fixme_should_propagate_errors();
-    return !strncmp(table->h.sig, signature.characters_without_null_termination(), 4);
+    auto check_in_region = [&](size_t length, auto&& callback) -> bool {
+        auto mapping_length = MUST(Memory::page_round_up(table_header.offset_in_page() + length));
```

Comment:
Maybe (ab)use `Memory::map_typed<u8 const>` here?

I know this might be a bit awkward, but we do this in other places as well and gets rid of the manual offset calculations.